index.js

```diff
@@ -10,125 +10,118 @@ import {
+let filterStatus;
+let searchResult;
+let currentPage = 1;
+const taskPerPage = 3;
+let totalPageCount;
+const startIndex = 0;
+
+export const isFilterTabDisable = () => {
+  todos?.length
+    ? tabs.forEach((tab) => {
+        tab.removeAttribute("disabled");
+      })
+    : tabs.forEach((tab) => {
+        tab.setAttribute("disabled", true);
+        tab.classList.remove("selected");
+      });
+};
 
-  tabs.forEach((tab) => tab.classList.remove("selected"));
+const filteredByPaginate = (todos) => {
+  totalPageCount = Math.ceil(todos.length / taskPerPage);
+  renderLoadMoreButton(todos);
+  return todos.slice(startIndex, taskPerPage * currentPage);
+};
 
-  event.target.classList.add("selected");
+const handleLoadMoreTask = () => {
+  if (todos && currentPage < totalPageCount) currentPage++;
+  renderTodoList(todos);
 };
 
-const renderTasksByStatus = (isDone) => {
-  keepCard1Elements();
-  filteredTodos = filteredTasks.length
-    ? filteredTasks.filter((todo) => todo.isDone === isDone)
-    : todos.filter((todo) => todo.isDone === isDone);
+export const renderLoadMoreButton = (todos) => {
+  if (todos.length >= taskPerPage && currentPage !== totalPageCount) {
+    return (
+      loadMoreButton.classList.remove("hide"),
+      loadMoreButton.classList.add("show")
+    );
+  }
 
-  filteredTodos.map((todo) => {
-    const newTask = createTaskElement(todo);
-    taskList.insertBefore(newTask, taskList.children[1]);
-  });
+  return (
+    loadMoreButton.classList.remove("show"),
+    loadMoreButton.classList.add("hide")
+  );
 };
 
-export const keepCard1Elements = () => {
-  const elementsToKeep = [];
-
-  for (let i = 0; i < taskList.children.length; i++) {
-    if (taskList.children[i].id === "card1") {
-      elementsToKeep.push(taskList.children[i]);
-    }
+const filteredByFilter = (status = "all", todos) => {
+  if (status === "all") {
+    return todos;
+  } else if (status === "complete") {
+    return todos?.filter((todo) => todo.isDone === true);
+  } else if (status === "incomplete") {
+    return todos?.filter((todo) => todo.isDone === false);
   }
+};
 
-  taskList.innerHTML = "";
-
-  elementsToKeep.map((element) => {
-    taskList.appendChild(element);
-  });
+const filteredBySearch = (searchText, todos) => {
+  return todos?.filter((task) => task.value.toLowerCase().includes(searchText));
 };
 
-const debounce = (handleSearch, delay) => {
+const debounce = (filteredBySearch, delay) => {
   let timer;
-
-  return function (...args) {
+  return function () {
     clearTimeout(timer);
-
-    timer = setTimeout(() => {
-      handleSearch(...args);
-    }, delay);
+    return new Promise((resolve) => {
+      timer = setTimeout(() => {
+        const result = filteredBySearch.apply(this, arguments);
+        resolve(result);
+      }, delay);
+    });
   };
 };
 
-const handleSearch = () => {
-  keepCard1Elements();
-
-  const searchText = searchInput.value.toLowerCase().trim();
+todos.length === 0 && isFilterTabDisable();
```

Comment:
By the name `isFilterTabDisable()` this function should return true of false. Why are we calling it here? If it doesn't return the state of filterTab but makes the filter tab disable, please rename it like so:
```suggestion
if(todos.length === 0){
  disableFilterBar()
}
```